contracts/contracts/poolBooster/PoolBoostCentralRegistry.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Governable } from "../governance/Governable.sol";
+
+/**
+ * @title Contract that holds all governance approved pool booster Factory
+ *        implementation deployments
+ * @author Origin Protocol Inc
+ */
+contract PoolBoostCentralRegistry is Governable {
+    event FactoryApproved(address factoryAddress);
+    event FactoryRemoved(address factoryAddress);
+
+    // @notice List of approved factories
+    address[] public factories;
+
+    constructor() {
+        // set the governor of the implementation contract to zero address
+        _setGovernor(address(0));
+    }
+
+    /**
+     * @notice Adds a factory address to the approved factory addresses
+     * @param _factoryAddress address of the factory
+     */
+    function approveFactory(address _factoryAddress) external onlyGovernor {
+        require(_factoryAddress != address(0), "Invalid address");
+
+        factories.push(_factoryAddress);
+        emit FactoryApproved(_factoryAddress);
+    }
```

Comment:
There is no check to ensure that the factory isn't already approved. 
But in the other hand, I don't see any issue if we have 2 times the same factory here. 
So I'm fine to keep it like this 👍

Reply:
Actually, that is a great point. Adding a check and a test here: https://github.com/OriginProtocol/origin-dollar/pull/2400/commits/735f55f18fae473b309d58b438569410822fbfe5

In case we have 2 entries of the same factory we could get a false sense of security from calling remove once. Since the other factory would still be in the list and considered supported.

Reply:
I didn't thought about this situation but indeed! Thanks for fixing it.